cdap-proto/src/main/java/co/cask/cdap/proto/id/ApplicationId.java

```diff
@@ -62,6 +63,12 @@ public String getEntityName() {
     return getApplication();
   }
 
+  @Override
+  public MetadataEntity toMetadataEntity() throws UnsupportedOperationException {
```

Comment:
`UnsupportedOperationException` is unchecked exception, no need to add it to `throws` clause.

Reply:
This is in APi from EntityId which is overridden here the method signature should remain same. Also regardless of exception is checked or unchecked its a good practice to explicitly mention it in public APIs

Reply:
hmm.. not sure if its true.. while providing method documentation using `@throws` yes you need all exceptions (checked and unchecked) to be documented.. however `throws` keyword should not include unchecked exceptions - https://books.google.com/books?id=ka2VUBqHiWkC&pg=PA252&lpg=PA252&dq=should+unchecked+exceptions+declared+in+throws#v=onepage&q=should%20unchecked%20exceptions%20declared%20in%20throws

Reply:
"Use the Javadoc @throws tag to document each unchecked exception that
a method can throw, but do not use the throws keyword to include unchecked
exceptions in the method declaration. " from the link I posted :)

Reply:
TLDR 😃